pkg/kube/pod_runner.go

```diff
@@ -45,11 +50,17 @@ func NewPodRunner(cli kubernetes.Interface, options *PodOptions) PodRunner {
 
 // Run will create a new Pod based on PodRunner contents and execute the given function
 func (p *podRunner) Run(ctx context.Context, fn func(context.Context, *v1.Pod) (map[string]interface{}, error)) (map[string]interface{}, error) {
+	return p.RunEx(ctx, func(ctx2 context.Context, pc PodController) (PodOutputMap, error) {
+		return fn(ctx2, pc.Pod())
```

Comment:
Nit. `ctx2` looks weird to me. I'd rename it to something more meaningful, e.g.:

```suggestion
	return p.RunEx(ctx, func(innerCtx context.Context, pc PodController) (PodOutputMap, error) {
		return fn(innerCtx, pc.Pod())
```